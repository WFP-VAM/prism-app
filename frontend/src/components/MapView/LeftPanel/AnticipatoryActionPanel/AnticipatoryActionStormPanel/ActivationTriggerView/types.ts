import { AACategory } from 'context/anticipatoryAction/AAStormStateSlice/parsedStormDataTypes';

export const AAPanelCategories: AACategory[] = [
  AACategory.Severe,
  AACategory.Moderate,
];

export const AADisplayCategory: {
  [key in AACategory]?: string;
} = {
  [AACategory.Severe]: ' >118 KM/H',
  [AACategory.Moderate]: ' >89 KM/H',
};
