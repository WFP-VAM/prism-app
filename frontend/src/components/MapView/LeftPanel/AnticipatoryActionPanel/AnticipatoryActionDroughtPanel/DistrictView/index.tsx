import {
  Button,
  Tooltip,
  Typography,
  createStyles,
  makeStyles,
} from '@material-ui/core';
import { lightGrey } from 'muiTheme';
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  AAView,
  AnticipatoryActionDataRow,
} from 'context/anticipatoryAction/AADroughtStateSlice/types';
import { AAWindowKeys } from 'config/utils';
import {
  AADataSelector,
  AAFiltersSelector,
  AASelectedDistrictSelector,
  setAASelectedDistrict,
  setAAView,
} from 'context/anticipatoryAction/AADroughtStateSlice';
import { useSafeTranslation } from 'i18n';
import { ClearAll, Equalizer, Reply } from '@material-ui/icons';
import { AADataSeverityOrder, getAAIcon } from '../utils';
import { useAACommonStyles } from '../../utils';
import {
  Action,
  getActionsByPhaseCategoryAndWindow,
} from './ActionsModal/actions';
import ActionsModal from './ActionsModal';
import { dateSorter, districtViewTransform } from './utils';

interface WindowColumnProps {
  win: (typeof AAWindowKeys)[number];
  transformed: ReturnType<typeof districtViewTransform>;
  rowKeys: string[];
  openActionsDialog: () => void;
  setModalActions: React.Dispatch<React.SetStateAction<Action[]>>;
}

function WindowColumn({
  win,
  transformed,
  rowKeys,
  openActionsDialog,
  setModalActions,
}: WindowColumnProps) {
  const classes = useWindowColumnStyles();
  const { t } = useSafeTranslation();

  const extendedTransformed = {
    ...Object.fromEntries(
      rowKeys.map(x => [x, [] as AnticipatoryActionDataRow[]]),
    ),
    ...(transformed?.transformed || {}),
  };

  const allEntries = Object.values(transformed?.transformed || {}).flat();

  if (!transformed) {
    return null;
  }
  const hasWindowData = Object.keys(transformed.months).length > 0;

  return (
    <div className={classes.windowWrapper}>
      <div style={{ textAlign: 'center' }}>
        <Typography variant="h4" className={classes.headerText}>
          {t(win)}
        </Typography>
      </div>
      {!hasWindowData && (
        <Typography variant="h3" className={classes.noDataText}>
          {t('No data')}
        </Typography>
      )}
      {hasWindowData && (
        <div className={classes.tableWrapper}>
          <div className={classes.headRowWrapper}>
            {Object.entries(transformed?.months || {}).map(x => (
              <div key={x[0]} className={classes.headColumn}>
                <Typography className={classes.monthText}>{t(x[1])}</Typography>
              </div>
            ))}
          </div>
          {
            // eslint-disable-next-line fp/no-mutating-methods
            Object.entries(extendedTransformed)
              .sort(dateSorter)
              .map(x => (
                <div key={x[0]} className={classes.rowWrapper}>
                  {Object.entries(transformed?.months || {}).map(y => {
                    const elem = x[1].find(z => z.date === y[0]);
                    if (!elem) {
                      return <div key={y[0]} className={classes.column} />;
                    }
                    return (
                      <div key={y[0]} className={classes.column}>
                        <div className={classes.iconWrapper}>
                          {getAAIcon(elem.category, elem.phase)}
                        </div>
                      </div>
                    );
                  })}
                </div>
              ))
          }
        </div>
      )}
      {hasWindowData && (
        <div className={classes.actionsWrapper}>
          <div style={{ textAlign: 'center' }}>
            <Typography variant="h3">{t('ACTIONS')}</Typography>
          </div>
          <div className={classes.actionBoxesWrapper}>
            {Object.keys(transformed.months).map(x => {
              const dateData = allEntries.filter(y => y.date === x);
              const elem = dateData.reduce(
                (max, curr) =>
                  AADataSeverityOrder(max.category, max.phase) >
                  AADataSeverityOrder(curr.category, curr.phase)
                    ? max
                    : curr,
                dateData[0],
              );
              const actions = getActionsByPhaseCategoryAndWindow(
                elem.phase,
                elem.category,
                elem.window,
              );
              return (
                <button
                  key={x}
                  type="button"
                  id={String(x)}
                  className={classes.actionBox}
                  style={{
                    justifyContent: actions.length <= 2 ? 'center' : undefined,
                    cursor: actions.length > 0 ? 'pointer' : undefined,
                  }}
                  onClick={() => {
                    if (actions.length === 0) {
                      return;
                    }
                    openActionsDialog();
                    setModalActions(actions);
                  }}
                >
                  {actions.map(action => (
                    // wrapping in div to show tooltip with FontAwesomeIcons
                    <Tooltip
                      key={action.name}
                      title={t(action.name) || action.name}
                    >
                      <div>{action.icon}</div>
                    </Tooltip>
                  ))}
                </button>
              );
            })}
          </div>
        </div>
      )}
    </div>
  );
}

const useWindowColumnStyles = makeStyles(() =>
  createStyles({
    windowWrapper: {
      display: 'flex',
      flexDirection: 'column',
      padding: '0rem 0.25rem',
    },
    tableWrapper: { display: 'flex', flexDirection: 'column', gap: '2px' },
    rowWrapper: {
      display: 'flex',
      flexDirection: 'row',
      minHeight: '3rem',
      background: 'white',
    },
    headRowWrapper: {
      display: 'flex',
      flexDirection: 'row',
      minHeight: '2.5rem',
      background: 'white',
    },
    headColumn: {
      width: '4.6rem',
      padding: '0.1rem 0.25rem',
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
    },
    column: {
      width: '4.6rem',
      height: '3.2rem',
      padding: '0.1rem 0.25rem',
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
    },
    headerText: {
      fontWeight: 'bold',
      textTransform: 'uppercase',
      height: '2rem',
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
    },
    monthText: {
      background: lightGrey,
      borderRadius: '4px',
      textAlign: 'center',
      textTransform: 'uppercase',
      lineHeight: '2rem',
      width: '100%',
    },
    iconWrapper: {
      width: '100%',
      height: '100%',
    },
    actionsWrapper: {
      display: 'flex',
      flexDirection: 'column',
      gap: '2px',
      padding: '0.5rem 0',
    },
    actionBoxesWrapper: {
      display: 'flex',
      flexDirection: 'row',
    },
    actionBox: {
      height: '6.2rem',
      width: '4.6rem',
      margin: '0.1rem 0.25rem',
      background: 'white',
      borderRadius: '4px',
      color: 'black',
      display: 'flex',
      flexWrap: 'wrap',
      flexDirection: 'column',
      alignItems: 'center',
      gap: '0.5rem',
      paddingTop: '0.8rem',
      paddingBottom: '0.8rem',
      border: 'none',
    },
    noDataText: {
      margin: 'auto',
    },
  }),
);

interface DistrictViewProps {
  dialogs: {
    text: string;
    onclick: () => void;
  }[];
}

function DistrictView({ dialogs }: DistrictViewProps) {
  const dispatch = useDispatch();
  const { t } = useSafeTranslation();
  const classes = useDistrictViewStyles();
  const commonClasses = useAACommonStyles();
  const { selectedWindow } = useSelector(AAFiltersSelector);
  const rawAAData = useSelector(AADataSelector);
  const aaFilters = useSelector(AAFiltersSelector);
  const selectedDistrict = useSelector(AASelectedDistrictSelector);
  const [actionsModalOpen, setActionsModalOpen] =
    React.useState<boolean>(false);
  const [modalActions, setModalActions] = React.useState<Action[]>([]);

  const districtButtons = [
    {
      icon: ClearAll,
      text: 'Timeline',
      onClick: () => dispatch(setAAView(AAView.Timeline)),
    },
    {
      icon: Equalizer,
      text: 'Forecast',
      onClick: () => dispatch(setAAView(AAView.Forecast)),
    },
    {
      icon: Reply,
      text: 'Summary',
      onClick: () =>
        dispatch(setAAView(AAView.Home)) && dispatch(setAASelectedDistrict('')),
    },
  ];

  const windows = selectedWindow === 'All' ? AAWindowKeys : [selectedWindow];
  const transformed = windows.map(x =>
    districtViewTransform(rawAAData[x][selectedDistrict], aaFilters),
  );
  const rowKeys = transformed
    .map(x => Object.keys(x?.transformed || {}))
    .flat();

  return (
    <>
      <div className={classes.root}>
        <ActionsModal
          open={actionsModalOpen}
          onClose={() => setActionsModalOpen(false)}
          actions={modalActions}
        />
        <div className={classes.districtViewWrapper}>
          {windows.map((win, index) => (
            <WindowColumn
              key={win}
              win={win}
              transformed={transformed[index]}
              rowKeys={rowKeys}
              openActionsDialog={() => setActionsModalOpen(true)}
              setModalActions={setModalActions}
            />
          ))}
        </div>
      </div>
      <div className={commonClasses.footerWrapper}>
        <div className={commonClasses.footerActionsWrapper}>
          {districtButtons.map(x => (
            <Button
              key={x.text}
              className={commonClasses.footerButton}
              variant="outlined"
              fullWidth
              onClick={x.onClick}
              startIcon={<x.icon />}
            >
              <Typography>{t(x.text)}</Typography>
            </Button>
          ))}
        </div>
        <div className={commonClasses.footerDialogsWrapper}>
          {dialogs.map(dialog => (
            <Typography
              key={dialog.text}
              className={commonClasses.footerDialog}
              component="button"
              onClick={() => dialog.onclick()}
            >
              {t(dialog.text)}
            </Typography>
          ))}
        </div>
      </div>
    </>
  );
}

const useDistrictViewStyles = makeStyles(() =>
  createStyles({
    root: {
      display: 'flex',
      flexDirection: 'column',
      width: '100%',
      background: lightGrey,
      overflow: 'scroll',
    },
    districtViewWrapper: {
      display: 'flex',
      flexDirection: 'row',
      width: '100%',
      background: lightGrey,
      overflow: 'scroll',
      justifyContent: 'space-around',
    },
  }),
);

export default DistrictView;
