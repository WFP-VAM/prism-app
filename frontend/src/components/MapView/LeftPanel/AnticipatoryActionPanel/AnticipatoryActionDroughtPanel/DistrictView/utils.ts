import {
  AnticipatoryActionDataRow,
  AnticipatoryActionState,
} from 'context/anticipatoryAction/AADroughtStateSlice/types';
import { getSeason } from 'context/anticipatoryAction/AADroughtStateSlice/utils';
import { AADataSeverityOrder } from '../utils';
import { TimelineRow } from '../Timeline/utils';

export function districtViewTransform(
  data: AnticipatoryActionDataRow[] | undefined,
  filters: AnticipatoryActionState['filters'],
) {
  if (!data) {
    return undefined;
  }
  const { categories: categoryFilters, selectedIndex, selectedDate } = filters;
  const season = getSeason(selectedDate);

  // Keep valid data only and switch SET phase to 'na' if the READY was not valid
  const validData = data
    .filter(
      x =>
        categoryFilters[x.category] &&
        season === x.season &&
        (selectedIndex === '' || x.index === selectedIndex),
    )
    .map(x => {
      if (x.isValid) {
        return x;
      }
      return {
        ...x,
        phase: 'na',
      } as AnticipatoryActionDataRow;
    });

  const monthsMap = new Map<string, AnticipatoryActionDataRow[]>();
  validData.forEach(x => {
    const val = monthsMap.get(x.date);
    monthsMap.set(x.date, val ? [...val, x] : [x]);
  });

  // eslint-disable-next-line fp/no-mutating-methods
  const months = Array.from(monthsMap.keys()).sort();

  const topFiltered = months
    .map(date => {
      const dateData = monthsMap.get(date);
      if (!dateData) {
        // this should never happen
        throw new Error('Invalid Date');
      }

      if (dateData.every(x => x.computedRow)) {
        return [];
      }

      const setCategories = new Set(
        dateData.filter(x => x.phase === 'Set').map(x => x.category),
      );

      const readyCategories = new Set(
        dateData.filter(x => x.phase === 'Ready').map(x => x.category),
      );

      // only keep data for the highest phase by category
      const ret = dateData.filter(
        x =>
          x.phase === 'Set' ||
          (x.phase === 'Ready' && !setCategories.has(x.category)) ||
          (x.phase === 'na' &&
            !setCategories.has(x.category) &&
            !readyCategories.has(x.category)),
      );

      return ret;
    })
    .flat();

  const sevMap = new Map<number, AnticipatoryActionDataRow[]>();
  topFiltered.forEach(x => {
    // return category data on one line by setting phase bonus to 0
    const sevVal = AADataSeverityOrder(x.category, x.phase, 0);
    const val = sevMap.get(sevVal);
    sevMap.set(sevVal, val ? [...val, x] : [x]);
  });

  const transformed = Object.fromEntries(sevMap);

  return {
    months: Object.fromEntries(
      [...new Set(topFiltered.map(x => x.date))].map(x => [
        x,
        new Date(x).toLocaleString('en-US', { month: 'short' }),
      ]),
    ),
    transformed,
  };
}

export function dateSorter(
  a: [string, AnticipatoryActionDataRow[] | TimelineRow],
  b: [string, AnticipatoryActionDataRow[] | TimelineRow],
) {
  if (a[0] > b[0]) {
    return -1;
  }
  if (a[0] < b[0]) {
    return 1;
  }
  return 0;
}
