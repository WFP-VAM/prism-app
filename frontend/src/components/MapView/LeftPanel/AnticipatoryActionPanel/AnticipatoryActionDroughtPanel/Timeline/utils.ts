import { AAWindowKeys } from 'config/utils';
import {
  AACategoryType,
  AAPhase,
  AAPhaseType,
  AAcategory,
  AnticipatoryActionDataRow,
  AnticipatoryActionState,
} from 'context/anticipatoryAction/AADroughtStateSlice/types';
import { getSeason } from 'context/anticipatoryAction/AADroughtStateSlice/utils';

function getColumnKey(val: AnticipatoryActionDataRow): number {
  const { category, phase } = val;
  const catIndex = AAcategory.findIndex(x => x === category);
  const phaseIndex = AAPhase.findIndex(x => x === phase);
  return catIndex * 10 + phaseIndex;
}

export type TimelineRow = {
  status: {
    category: AACategoryType;
    phase: AAPhaseType;
  };
  data: AnticipatoryActionDataRow[];
};

interface TimelineTransformParams {
  filters: AnticipatoryActionState['filters'];
  selectedDistrict: AnticipatoryActionState['selectedDistrict'];
  data: AnticipatoryActionState['data'];
}

export function timelineTransform({
  filters,
  selectedDistrict,
  data,
}: TimelineTransformParams) {
  const { selectedWindow, selectedIndex, categories, selectedDate } = filters;
  const season = getSeason(selectedDate);

  const windowData = (
    selectedWindow === 'All' ? AAWindowKeys : [selectedWindow]
  ).map(win => {
    const districtData = !!selectedDistrict && data[win][selectedDistrict];
    if (!districtData) {
      return [win, null];
    }

    const filtered = districtData.filter(
      x =>
        x.season === season &&
        !x.computedRow &&
        (selectedIndex === '' || selectedIndex === x.index) &&
        categories[x.category],
    );

    const months = [...new Set(filtered.map(x => x.date))].map(x => [
      x,
      new Date(x).toLocaleString('en-US', { month: 'short' }),
    ]);

    const categoriesMap = new Map<
      number,
      {
        status: { category: AACategoryType; phase: AAPhaseType };
        data: AnticipatoryActionDataRow[];
      }
    >();
    filtered.forEach(x => {
      const key = getColumnKey(x);
      const val = categoriesMap.get(key);
      if (val) {
        // Sort the new data based on validity
        // eslint-disable-next-line fp/no-mutating-methods
        const sortedData = [
          x,
          ...val.data.filter(item => item.date === x.date),
        ].sort((a, b) => {
          if (a.isValid && a.isOtherPhaseValid) {
            return -1;
          }
          if (b.isValid && b.isOtherPhaseValid) {
            return 1;
          }
          if (a.isValid) {
            return -1;
          }
          if (b.isValid) {
            return 1;
          }
          return 0;
        });

        // Keep only the highest priority item for this date
        const highestPriorityItem = sortedData[0];

        // Update the data array
        // eslint-disable-next-line fp/no-mutation
        val.data = [
          ...val.data.filter(item => item.date !== x.date),
          highestPriorityItem,
        ];
      } else {
        categoriesMap.set(key, {
          status: {
            category: x.category,
            phase: x.phase,
          },
          data: [x],
        });
      }
    });
    return [win, { months, rows: Object.fromEntries(categoriesMap) }];
  }) as [
    (typeof AAWindowKeys)[number],
    {
      months: string[][];
      rows: { [id: string]: TimelineRow };
    } | null,
  ][];

  const allRows = windowData
    .map(([_win, x]) =>
      Object.entries(x?.rows || {}).map(([id, info]) => [
        id,
        { status: info.status, data: [] },
      ]),
    )
    .flat() as [string, TimelineRow][];

  return {
    windowData: Object.fromEntries(windowData),
    allRows: Object.fromEntries(allRows),
  };
}
