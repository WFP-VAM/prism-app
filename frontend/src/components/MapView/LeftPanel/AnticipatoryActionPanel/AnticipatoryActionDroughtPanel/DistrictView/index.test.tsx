import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { AnticipatoryActionState } from 'context/anticipatoryAction/AADroughtStateSlice/types';
import { Panel } from 'config/types';
import DistrictView from '.';
import { defaultDialogs, mockAAData } from '../test.utils';
import { districtViewTransform } from './utils';

const mockStore = configureStore([]);

const filters: AnticipatoryActionState['filters'] = {
  selectedWindow: 'Window 1',
  categories: {
    Severe: true,
    Moderate: true,
    Normal: true,
    Mild: true,
    na: true,
    ny: true,
  },
  selectedIndex: '',
  selectedDate: '2023-05-01',
};

const store = mockStore({
  mapState: {
    layers: [],
    dateRange: { startDate: 1701432000000 },
    maplibreMap: () => {},
    errors: [],
    layersData: [],
    loadingLayerIds: [],
    boundaryRelationData: {},
  },
  serverState: { availableDates: {}, loading: false },
  anticipatoryActionDroughtState: {
    filters,
    selectedDistrict: 'Changara',
    monitoredDistricts: ['Changara'],
    data: mockAAData,
  },
  leftPanelState: {
    tabValue: Panel.AnticipatoryActionDrought,
  },
});

test('renders as expected', () => {
  const { container } = render(
    <BrowserRouter>
      <Provider store={store}>
        <DistrictView dialogs={defaultDialogs} />
      </Provider>
    </BrowserRouter>,
  );
  expect(container).toMatchSnapshot();
});

test('District view transformation', () => {
  const res = districtViewTransform(mockAAData['Window 1'].Changara, filters);

  expect(res).toEqual(out);
});

const out = {
  months: {
    '2023-08-01': 'Aug',
    '2023-09-01': 'Sep',
    '2023-10-01': 'Oct',
    '2023-11-01': 'Nov',
  },
  transformed: {
    '30': [
      {
        category: 'Mild',
        date: '2023-08-01',
        season: '2023-24',
        district: 'Changara',
        index: 'SPI DJF',
        isValid: false,
        isOtherPhaseValid: false,
        new: false,
        phase: 'na',
        probability: 0.16,
        trigger: 0.17,
        type: 'SPI',
        window: 'Window 1',
      },
      {
        category: 'Mild',
        date: '2023-09-01',
        season: '2023-24',
        district: 'Changara',
        index: 'SPI DJF',
        isValid: false,
        isOtherPhaseValid: false,
        new: false,
        phase: 'na',
        probability: 0.22,
        trigger: 0.3,
        type: 'SPI',
        window: 'Window 1',
      },
      {
        category: 'Mild',
        date: '2023-09-01',
        season: '2023-24',
        district: 'Changara',
        index: 'SPI DJF',
        isValid: false,
        isOtherPhaseValid: false,
        new: false,
        phase: 'na',
        probability: 0.12,
        trigger: 0.3,
        type: 'SPI',
        window: 'Window 1',
      },
      {
        category: 'Mild',
        district: 'Changara',
        index: 'SPI DJF',
        type: 'SPI',
        window: 'Window 1',
        new: false,
        phase: 'Ready',
        probability: 0.31,
        trigger: 0.2,
        date: '2023-10-01',
        season: '2023-24',
        isValid: true,
        isOtherPhaseValid: true,
      },
      {
        category: 'Mild',
        district: 'Changara',
        index: 'SPI DJF',
        type: 'SPI',
        window: 'Window 1',
        new: true,
        phase: 'Set',
        probability: 0.28,
        trigger: 0.25,
        date: '2023-11-01',
        season: '2023-24',
        isValid: true,
        isOtherPhaseValid: true,
      },
    ],
    '40': [
      {
        category: 'Moderate',
        district: 'Changara',
        index: 'SPI DJF',
        type: 'SPI',
        window: 'Window 1',
        new: true,
        phase: 'Ready',
        probability: 0.19,
        trigger: 0.1,
        date: '2023-09-01',
        season: '2023-24',
        isValid: true,
        isOtherPhaseValid: false,
      },
      {
        category: 'Moderate',
        district: 'Changara',
        index: 'SPI DJF',
        type: 'SPI',
        window: 'Window 1',
        new: false,
        phase: 'Ready',
        probability: 0.28,
        trigger: 0.2,
        date: '2023-10-01',
        season: '2023-24',
        isValid: true,
        isOtherPhaseValid: false,
      },
      {
        category: 'Moderate',
        district: 'Changara',
        index: 'SPI DJF',
        type: 'SPI',
        window: 'Window 1',
        new: false,
        phase: 'na',
        probability: 0.22,
        trigger: 0.3,
        date: '2023-11-01',
        season: '2023-24',
        isValid: false,
        isOtherPhaseValid: true,
      },
    ],
  },
};
