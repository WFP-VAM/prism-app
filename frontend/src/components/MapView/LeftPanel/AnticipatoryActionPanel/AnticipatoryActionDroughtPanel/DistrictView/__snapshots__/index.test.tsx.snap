// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <mock-dialog
      open="false"
    >
      <mock-dialogtitle>
        <div
          class="makeStyles-titleWrapper-12"
        >
          <mock-typography
            variant="h2"
          >
            Recommended actions
          </mock-typography>
          <button
            class="MuiButtonBase-root MuiIconButton-root"
            style="padding: 0px;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </mock-dialogtitle>
      <mock-dialogcontent
        dividers="true"
      >
        <div
          class="makeStyles-contentWrapper-13"
        />
      </mock-dialogcontent>
      <mock-dialogactions
        classname="makeStyles-dialogActionsWrapper-16"
      >
        <mock-button
          classname="makeStyles-dialogButton-17"
          starticon="[object Object]"
          variant="outlined"
        >
          <mock-typography>
            Close
          </mock-typography>
        </mock-button>
      </mock-dialogactions>
    </mock-dialog>
    <div
      class="makeStyles-districtViewWrapper-2"
    >
      <div
        class="makeStyles-windowWrapper-18"
      >
        <div
          style="text-align: center;"
        >
          <mock-typography
            classname="makeStyles-headerText-24"
            variant="h4"
          >
            Window 1
          </mock-typography>
        </div>
        <div
          class="makeStyles-tableWrapper-19"
        >
          <div
            class="makeStyles-headRowWrapper-21"
          >
            <div
              class="makeStyles-headColumn-22"
            >
              <mock-typography
                classname="makeStyles-monthText-25"
              >
                Aug
              </mock-typography>
            </div>
            <div
              class="makeStyles-headColumn-22"
            >
              <mock-typography
                classname="makeStyles-monthText-25"
              >
                Sep
              </mock-typography>
            </div>
            <div
              class="makeStyles-headColumn-22"
            >
              <mock-typography
                classname="makeStyles-monthText-25"
              >
                Oct
              </mock-typography>
            </div>
            <div
              class="makeStyles-headColumn-22"
            >
              <mock-typography
                classname="makeStyles-monthText-25"
              >
                Nov
              </mock-typography>
            </div>
          </div>
          <div
            class="makeStyles-rowWrapper-20"
          >
            <div
              class="makeStyles-column-23"
            />
            <div
              class="makeStyles-column-23"
            >
              <div
                class="makeStyles-iconWrapper-26"
              >
                <div
                  class="makeStyles-iconWrapper-31"
                  style="background: rgb(255, 213, 45);"
                >
                  <div
                    class="makeStyles-centerContainer-32"
                    style="border: 1px solid black; color: black;"
                  >
                    <div
                      class="makeStyles-topTextContainer-33"
                      style="border-bottom: 1px solid black; background: rgb(255, 213, 45);"
                    >
                      R
                    </div>
                    <div
                      class="makeStyles-bottomTextContainer-34"
                      style="background: rgb(255, 213, 45);"
                    >
                      MOD
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="makeStyles-column-23"
            >
              <div
                class="makeStyles-iconWrapper-26"
              >
                <div
                  class="makeStyles-iconWrapper-31"
                  style="background: rgb(255, 213, 45);"
                >
                  <div
                    class="makeStyles-centerContainer-32"
                    style="border: 1px solid black; color: black;"
                  >
                    <div
                      class="makeStyles-topTextContainer-33"
                      style="border-bottom: 1px solid black; background: rgb(255, 213, 45);"
                    >
                      R
                    </div>
                    <div
                      class="makeStyles-bottomTextContainer-34"
                      style="background: rgb(255, 213, 45);"
                    >
                      MOD
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="makeStyles-column-23"
            >
              <div
                class="makeStyles-iconWrapper-26"
              >
                <div
                  class="makeStyles-iconWrapper-31"
                  style="background: rgb(241, 241, 241);"
                >
                  <div
                    class="makeStyles-centerContainer-32"
                    style="border: 1px solid black; color: black;"
                  >
                    <div
                      class="makeStyles-topTextContainer-33"
                      style="border-bottom: 1px solid black; background: rgb(241, 241, 241);"
                    >
                      na
                    </div>
                    <div
                      class="makeStyles-bottomTextContainer-34"
                      style="background: rgb(241, 241, 241);"
                    >
                      MOD
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="makeStyles-rowWrapper-20"
          >
            <div
              class="makeStyles-column-23"
            >
              <div
                class="makeStyles-iconWrapper-26"
              >
                <div
                  class="makeStyles-iconWrapper-31"
                  style="background: rgb(241, 241, 241);"
                >
                  <div
                    class="makeStyles-centerContainer-32"
                    style="border: 1px solid black; color: black;"
                  >
                    <div
                      class="makeStyles-topTextContainer-33"
                      style="border-bottom: 1px solid black; background: rgb(241, 241, 241);"
                    >
                      na
                    </div>
                    <div
                      class="makeStyles-bottomTextContainer-34"
                      style="background: rgb(241, 241, 241);"
                    >
                      MIL
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="makeStyles-column-23"
            >
              <div
                class="makeStyles-iconWrapper-26"
              >
                <div
                  class="makeStyles-iconWrapper-31"
                  style="background: rgb(241, 241, 241);"
                >
                  <div
                    class="makeStyles-centerContainer-32"
                    style="border: 1px solid black; color: black;"
                  >
                    <div
                      class="makeStyles-topTextContainer-33"
                      style="border-bottom: 1px solid black; background: rgb(241, 241, 241);"
                    >
                      na
                    </div>
                    <div
                      class="makeStyles-bottomTextContainer-34"
                      style="background: rgb(241, 241, 241);"
                    >
                      MIL
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="makeStyles-column-23"
            >
              <div
                class="makeStyles-iconWrapper-26"
              >
                <div
                  class="makeStyles-iconWrapper-31"
                  style="background: rgb(255, 252, 179);"
                >
                  <div
                    class="makeStyles-centerContainer-32"
                    style="border: 1px solid black; color: black;"
                  >
                    <div
                      class="makeStyles-topTextContainer-33"
                      style="border-bottom: 1px solid black; background: rgb(255, 252, 179);"
                    >
                      R
                    </div>
                    <div
                      class="makeStyles-bottomTextContainer-34"
                      style="background: rgb(255, 252, 179);"
                    >
                      MIL
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="makeStyles-column-23"
            >
              <div
                class="makeStyles-iconWrapper-26"
              >
                <div
                  class="makeStyles-iconWrapper-31"
                  style="background: rgb(255, 245, 3);"
                >
                  <div
                    class="makeStyles-centerContainer-32"
                    style="border: 1px solid black; color: black;"
                  >
                    <div
                      class="makeStyles-topTextContainer-33"
                      style="border-bottom: 1px solid black; background: rgb(255, 245, 3);"
                    >
                      S
                    </div>
                    <div
                      class="makeStyles-bottomTextContainer-34"
                      style="background: rgb(255, 245, 3);"
                    >
                      MIL
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="makeStyles-actionsWrapper-27"
        >
          <div
            style="text-align: center;"
          >
            <mock-typography
              variant="h3"
            >
              ACTIONS
            </mock-typography>
          </div>
          <div
            class="makeStyles-actionBoxesWrapper-28"
          >
            <button
              class="makeStyles-actionBox-29"
              id="2023-08-01"
              style="justify-content: center; cursor: pointer;"
              type="button"
            >
              <mock-tooltip
                title="No anticipatory action linked to Mild levels"
              >
                <div>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    style="font-size: 1rem; color: darkgrey;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 13H5v-2h14v2z"
                    />
                  </svg>
                </div>
              </mock-tooltip>
            </button>
            <button
              class="makeStyles-actionBox-29"
              id="2023-09-01"
              style="justify-content: center; cursor: pointer;"
              type="button"
            >
              <mock-tooltip
                title="Launch tenders and other procurement procedures"
              >
                <div>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    style="font-size: 1rem;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M22 11V3h-7v3H9V3H2v8h7V8h2v10h4v3h7v-8h-7v3h-2V8h2v3z"
                    />
                  </svg>
                </div>
              </mock-tooltip>
            </button>
            <button
              class="makeStyles-actionBox-29"
              id="2023-10-01"
              style="justify-content: center; cursor: pointer;"
              type="button"
            >
              <mock-tooltip
                title="Launch tenders and other procurement procedures"
              >
                <div>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    style="font-size: 1rem;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M22 11V3h-7v3H9V3H2v8h7V8h2v10h4v3h7v-8h-7v3h-2V8h2v3z"
                    />
                  </svg>
                </div>
              </mock-tooltip>
            </button>
            <button
              class="makeStyles-actionBox-29"
              id="2023-11-01"
              style="justify-content: center; cursor: pointer;"
              type="button"
            >
              <mock-tooltip
                title="No anticipatory action linked to Mild levels"
              >
                <div>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    style="font-size: 1rem; color: darkgrey;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 13H5v-2h14v2z"
                    />
                  </svg>
                </div>
              </mock-tooltip>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="makeStyles-footerWrapper-3"
  >
    <div
      class="makeStyles-footerActionsWrapper-4"
    >
      <mock-button
        classname="makeStyles-footerButton-6"
        fullwidth="true"
        starticon="[object Object]"
        variant="outlined"
      >
        <mock-typography>
          Timeline
        </mock-typography>
      </mock-button>
      <mock-button
        classname="makeStyles-footerButton-6"
        fullwidth="true"
        starticon="[object Object]"
        variant="outlined"
      >
        <mock-typography>
          Forecast
        </mock-typography>
      </mock-button>
      <mock-button
        classname="makeStyles-footerButton-6"
        fullwidth="true"
        starticon="[object Object]"
        variant="outlined"
      >
        <mock-typography>
          Summary
        </mock-typography>
      </mock-button>
    </div>
    <div
      class="makeStyles-footerDialogsWrapper-5"
    >
      <mock-typography
        classname="makeStyles-footerDialog-7"
        component="button"
      >
        How to read this screen
      </mock-typography>
    </div>
  </div>
</div>
`;
